bundles/org.openhab.binding.bondhome/src/main/java/org/openhab/binding/bondhome/internal/handler/BondBridgeHandler.java

```diff
@@ -0,0 +1,299 @@
+    private final BPUPListener udpListener;
+    private final BondHttpApi api;
+
+    private @NonNullByDefault({}) BondBridgeConfiguration config;
+
+    private final Set<BondDeviceHandler> handlers = Collections.synchronizedSet(new HashSet<>());
+
+    public BondBridgeHandler(Bridge bridge) {
+        super(bridge);
+        udpListener = new BPUPListener(this);
+        api = new BondHttpApi(this);
+        logger.debug("Created a BondBridgeHandler for thing '{}'", getThing().getUID());
+    }
+
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+        // Not needed, all commands are handled in the {@link BondDeviceHandler}
+    }
+
+    @Override
+    public void initialize() {
+        logger.debug("Start initializing the Bond bridge!");
+        config = getConfigAs(BondBridgeConfiguration.class);
+
+        // set the thing status to UNKNOWN temporarily
+        updateStatus(ThingStatus.UNKNOWN);
+
+        // Example for background initialization:
+        scheduler.execute(() -> {
+            initializeThing();
+        });
+    }
+
+    private void initializeThing() {
+        config = getConfigAs(BondBridgeConfiguration.class);
+        if (config != null) {
+            if (config.bondIpAddress == null) {
+                try {
+                    logger.trace("IP address of Bond {} is unknown", config.bondId);
+                    String lookupAddress = config.bondId + ".local";
+                    logger.trace("Attempting to get IP address for Bond Bridge {}", lookupAddress);
+                    InetAddress ia = InetAddress.getByName(lookupAddress);
+                    String ip = ia.getHostAddress();
+                    Configuration c = editConfiguration();
+                    c.put(CONFIG_IP_ADDRESS, ip);
+                    updateConfiguration(c);
+                } catch (UnknownHostException ignored) {
+                    updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR,
+                            "Unable to get an IP Address for Bond Bridge");
+                    return;
+                }
+            } else {
+                try {
+                    InetAddress.getByName(config.bondIpAddress);
+                } catch (UnknownHostException ignored) {
+                    updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR,
+                            "IP Address or host name for Bond Bridge is not valid");
+                }
+            }
+        }
+
+        // Ask the bridge it's current status and update the properties with the info
+        // This will also set the thing status to online/offline based on whether it
+        // succeeds in getting the properties from the bridge.
+        updateBridgeProperties();
+
+        // Finish
+        logger.debug("Finished initializing Bond bridge!");
+    }
+
+    @Override
+    public void dispose() {
+        logger.trace("Disposing Bond Bridge handler");
```

Comment:
Can this message be replaced by using the debugger or by increasing the framework's log level?

Reply:
removed